client/image_utils.go

```diff
@@ -95,3 +97,24 @@ func GetConfigSyncImageName() string {
 func GetConfigSyncImagePullPolicy() string {
 	return getPullPolicy(ConfigSyncPullPolicyEnvKey)
 }
+
+func SetImageRegistry(url *url.URL) error {
+
+	path := []string{url.String(), RouterImageName}
+	err := os.Setenv(RouterImageEnvKey, strings.Join(path, "/"))
```

Comment:
Instead of setting the individual image values if the registry is specified, what about changing the implementation of the various GetXXXImageName() methods to check if an env var setting the registry has been set (only if the full image name is not already specified) and use that if so?  

If wanted you can still keep the cli option, and it would just set that new env var for registry. (Personally I think I would perhaps let users set the registry env var themselves and not try to do it through the cli options).

Reply:
Agreed. That would then simply change what the "default" value is, if the existing env vars are not set, right?

Reply:
Exactly.

Reply:
Agreed. @ajssmith what do you think about removing the cli option? Is it possible that users are already used to modify the current env vars? In that case, maybe new users would need some kind of awareness of these env vars through the `help` command?